### PR TITLE
Devise 4.3.0 installation

### PR DESCRIPTION
**Pull request by sanchez2210**

### CODE REVIEW

**Comment by pachonk** [comment]
@sanchez2210 Looking to also have the user model gen'd please

**Comment by pachonk** [comment]
Did you leave these configs 100% vanilla?

**Comment by sanchez2210** [comment]
Yes, it's devise as rails g devise:install leaves it.

File: db/migrate/20170703044736_devise_create_users.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+      ## Trackable
+      t.integer  :sign_in_count, default: 0, null: false
+      t.datetime :current_sign_in_at
+      t.datetime :last_sign_in_at
+      t.inet     :current_sign_in_ip
+      t.inet     :last_sign_in_ip
+
+      ## Confirmable
+      # t.string   :confirmation_token
```
Comment (pachonk) [inline db/migrate/20170703044736_devise_create_users.rb]: Let's turn on confirmable and lockable right away.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+ 
+RSpec.describe User, type: :model do
+  it 'has a valid factory' do
+    expect(FactoryGirl.build(:user)).to be_valid
+  end
+end
```
Comment (pachonk) [inline spec/models/user_spec.rb]: New line at EOF.

File: db/migrate/20170703044736_devise_create_users.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      t.string   :unlock_token # Only if unlock strategy is :email or :both
+      t.datetime :locked_at
+
+
+      t.timestamps null: false
+    end
+
+    add_index :users, :email,                unique: true
+    add_index :users, :reset_password_token, unique: true
+    # add_index :users, :confirmation_token,   unique: true
```
Comment (pachonk) [inline db/migrate/20170703044736_devise_create_users.rb]: Have to add these indexes

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
```
Comment (pachonk) [inline app/models/user.rb]: and add them to this list, too

Comment (Exellin) [inline app/models/user.rb]: still need a new line at end of file here.

Comment (pachonk) [inline app/models/user.rb]: There is a line at EOF here.
There would be a red marker if there wasn't.

**Comment by Exellin** [comment]
My only problems were the lack of EOF lines in your ruby files but those can be easily added at a later date.

**Comment by pachonk** [comment]
@Exellin There would be a red icon if there wasn't a new line at EOF. The files are fine :)

**Comment by Exellin** [comment]
oh that's what that icon means.  I was looking for a literal blank line in the files.  my bad.

**Comment by troystarwalt** [comment]
xI'm getting an error when creating users that links back to not setting a default_host with devise. Anyone else seeing this?

`config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }`

**Comment by pachonk** [comment]
@troystarwalt Why don't you make a brand new issue? Generally comments on merged/closed issues and PRs can get lost fairly easily.

**Comment by sanchez2210** [comment]
@troystarwalt indeed, I left it like that. Perfect reason for a new issue.

**Pull request merged**